exercises/runLengthEncode/runLengthEncode.js

```diff
@@ -20,11 +20,24 @@
  * @returns {string} A run-length encoded copy of the input string
  */
 
-function runLengthEncode(num) {
-  // This is your job. :)
-
-  // Remember, if the code is stumping you, take a step back and
-  // make sure you can do it by hand.
+function runLengthEncode(string) {
```

Comment:
Consider:

```javascript
function runLengthEncode(string) {
  let count = 1;
  let prev = string[0];
  let result = '';

  for (let i = 1; i < string.length; i += 1) {
    if (string[i] !== prev) {
      result += String(count) + prev;

      prev = string[i];
      count = 1;
    } else {
      count += 1;
    }
  }

  result += String(count) + prev;

  return result;
}
```

Reply:
Compare my implementation above with your using the benchmark code from the sessions repository. See if you can't figure out what's going on.